src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -432,7 +505,11 @@ void EIRPlantLoopHeatPump::calcAvailableCapacity(EnergyPlusData &state, Real64 c
     }
 
     // check the curve values, reset to zero if negative
-    this->capModFTCurveCheck(state, loadSideOutletSetpointTemp, capacityModifierFuncTemp);
+    if (this->heatRecoveryIsActive && this->heatRecoveryCapFTempCurveIndex > 0) {
+        this->heatRecoveryCapModFTCurveCheck(state, loadSideOutletSetpointTemp, capacityModifierFuncTemp);
+    } else {
+        this->capModFTCurveCheck(state, loadSideOutletSetpointTemp, capacityModifierFuncTemp);
+    }
```

Comment:
Capacity modifier curve value check depending on the curve types specified and the status of the heat recovery operation.